subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/fixtures/TransformFixture.groovy

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.configurationcache.fixtures
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.transform.InputArtifact
+import org.gradle.api.artifacts.transform.TransformAction
+import org.gradle.api.artifacts.transform.TransformOutputs
+import org.gradle.api.artifacts.transform.TransformParameters
+import org.gradle.api.artifacts.type.ArtifactTypeDefinition
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.FileSystemLocation
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.TaskAction
+import org.gradle.test.fixtures.file.TestFile
+
+import java.nio.file.Files
+import java.util.stream.Collectors
+
+/**
+ * A helper to define a plugin that applies a simple transform action to dependencies.
+ */
+class TransformFixture {
```

Comment:
Any reason not to use/extend `ArtifactTransformTestFixture`?

Reply:
> Any reason not to use/extend `ArtifactTransformTestFixture`?

Mostly because I overlooked it when doing the implementation. I'll try to utilize it in a follow-up PR. Thank you for pointing me at it.